### PR TITLE
Rewrite script update loop

### PR DESCRIPTION
**Pull request by mindphluxnet**
Yep, turns out deleting from a list while iterating over it isn't the best idea. I rewrote it to be safe.

### CODE REVIEW

**Comment by GrenderG** [comment]
Seems like there's also an issue with `creature_ai_script_get_by_id`. With the current implementation, only 1 event per script is being taken into account, but a script can have multiple events.

File: game/world/managers/objects/script/ScriptHandler.py [hunk 1]
```diff
@@ -195,6 +187,12 @@ def update(self):
 
                 self.ooc_running = False
 
+        for i, script in enumerate(self.script_queue):
```
Comment (devw4r) [inline game/world/managers/objects/script/ScriptHandler.py:190]: Better loop over the collection copy, for script in list(self.script_queue):

File: game/world/managers/objects/script/ScriptHandler.py [hunk 2]
```diff
@@ -195,6 +187,12 @@ def update(self):
...
+            if time.time() - script.time_added >= script.delay:
+                ScriptHandler.handle_script(self, script)
+                del self.script_queue[i]
+                break
```
Comment (devw4r) [inline game/world/managers/objects/script/ScriptHandler.py:194]: break would short circuit, is that intended?

**Comment by mindphluxnet** [comment]
Something is indeed off with the OOC scripts but I'll leave this for somebody else to code.

**Pull request closed without merging**